gather_gitops.sh

```diff
@@ -55,144 +143,281 @@ function getNamespaces() {
-    oc get events -n openshift-gitops --field-selector type=Warning  > "${GITOPS_DIR}"/namespace_"${namespace}"_resources/warning-events.txt
-
-    echo "getting error events in ${namespace}..."
-    oc get events -n openshift-gitops --field-selector type=Error > "${GITOPS_DIR}"/namespace_"${namespace}"_resources/error-events.txt
-
-    # getting logs
+    echo " * Getting warning events in ${namespace}..."
+    EVENTS_DIR="${RESOURCES_DIR}/events"
+    create_directory "${EVENTS_DIR}"
+    run_and_log "oc get events -n ${namespace} --field-selector type=Warning" "${EVENTS_DIR}/warning-events.txt"
+    echo " * Getting error events in ${namespace}..."
+    run_and_log "oc get events -n ${namespace} --field-selector type=Error" "${EVENTS_DIR}/error-events.txt"
+    echo " * Getting all events in ${namespace}..."
+    run_and_log "oc get events -n ${namespace}" "${EVENTS_DIR}/all-events.txt"
+
+    echo " * Getting ArgoCD logs in ${namespace}..."
+    ARGOCD_LOG_DIR="${ARGOCD_DIR}/logs"
+    create_directory "${ARGOCD_LOG_DIR}"
     local argoCDName
     argoCDName=$(oc -n "${namespace}" get argocd -o jsonpath='{.items[*].metadata.name}')
-    oc logs statefulset/"${argoCDName}"-application-controller -n "${namespace}"  > "${GITOPS_DIR}"/namespace_"${namespace}"_resources/application_controller_logs.txt
-    oc logs deployment/"${argoCDName}"-server -n "${namespace}" > "${GITOPS_DIR}"/namespace_"${namespace}"_resources/server_logs.txt
-    oc logs deployment/"${argoCDName}"-repo-server -n "${namespace}" > "${GITOPS_DIR}"/namespace_"${namespace}"_resources/repo-server_logs.txt
-    oc logs deployment/"${argoCDName}"-redis -n "${namespace}" > "${GITOPS_DIR}"/namespace_"${namespace}"_resources/redis_logs.txt
-    oc logs deployment/"${argoCDName}"-dex-server -n "${namespace}" > "${GITOPS_DIR}"/namespace_"${namespace}"_resources/dex-server_logs.txt
+    run_and_log "oc logs statefulset/${argoCDName}-application-controller -n ${namespace}" "${ARGOCD_LOG_DIR}/application-controller-logs.txt"
+    run_and_log "oc logs deployment/${argoCDName}-server -n ${namespace}" "${ARGOCD_LOG_DIR}/server-logs.txt"
+    run_and_log "oc logs deployment/${argoCDName}-repo-server -n ${namespace}" "${ARGOCD_LOG_DIR}/repo-server-logs.txt"
+    run_and_log "oc logs deployment/${argoCDName}-redis -n ${namespace}" "${ARGOCD_LOG_DIR}/redis-logs.txt"
+    run_and_log "oc logs deployment/${argoCDName}-dex-server -n ${namespace}" "${ARGOCD_LOG_DIR}/dex-server-logs.txt"
   
+    echo " * Getting ArgoCD Managed namespaces in ${namespace}..."
+    run_and_log "oc get namespaces --selector=argocd.argoproj.io/managed-by=${namespace}" "${ARGOCD_DIR}/managed-namespaces.txt"
     local managedNamespaces
     managedNamespaces=$(oc get namespaces --selector=argocd.argoproj.io/managed-by="${namespace}" -o jsonpath='{.items[*].metadata.name}')
 
     for managedNamespace in ${managedNamespaces}; do
+      MANAGED_RESOURCES_DIR="${GITOPS_DIR}/managedNamespace_${managedNamespace}/resources"
```

Comment:
do you want to keep the `resources` folder as part of this if there aren't any other children folders to `"${GITOPS_DIR}/managedNamespace_${managedNamespace}/`?
Also, did you mean to make the managed namesources folder at the same level as all other namespaces? I originally had it so that the managed namespace folder was a child of the namespace folder that it was managed by, but if this makes more sense to you we can also keep it at this level.

Reply:
I have no strong preference. Basically I don't know what these managed namesources, so I might did that by accident. Which directory you would like me to put them?

Reply:
I had originally had them going to `"${GITOPS_DIR}/namespace_${namespace}_resources/managedNamespace_${managedNamespace}/"`, so I guess you could put them in `"${RESOURCES_DIR}/managedNamespace_${managedNamespace}/"`

Reply:
Ok. i've put them there now :)